tpm/aziot-tpm-sys/build.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+#![deny(rust_2018_idioms)]
+#![warn(clippy::all, clippy::pedantic)]
+
+use std::env;
+use std::path::Path;
+use std::process::Command;
+
+use cmake::Config;
+
+const SSL_OPTION: &str = "use_openssl";
+const USE_EMULATOR: &str = "use_emulator";
+
+trait SetPlatformDefines {
```

Comment:
Why make a trait with just one implementation instead of a free function?

Reply:
As with many things in this file, it's a remnant from when Windows support was a thing.

That said, having it as a trait _does_ make things a bit cleaner, as it plays nicely with the other `.define()` chains down below.

I see not real harm leaving it as an extension trait.